### PR TITLE
implement recipe edit through Composer

### PR DESCRIPTION
**Pull request by MyMrMush**

### CODE REVIEW

File: app/backend/recipes/src/main/java/com/juotava/recipes/controller/RecipesController.java
Comment (vans1646) [inline app/backend/recipes/src/main/java/com/juotava/recipes/controller/RecipesController.java:1]: Are there currently other places that use this endpoint? Not sure right now but what about the browser, if a user clicks a recipe there they don't have to be logged in right? So here authentication.getName() would not be possible
Could you please clarify if that's the case?

File: app/frontend/src/components/composer/Composer.js [hunk 1]
```diff
@@ -79,7 +81,7 @@ function Composer() {
         tmp.draft = draft;
         tmp.createdBy = user.sub;
         setRecipe((prevRecipe) => ({...prevRecipe, draft: draft, createdBy: user.sub}));
-        console.log('Saving Recipe:', tmp);
+        //console.log('Saving Recipe:', tmp);
```
Comment (vans1646) [inline app/frontend/src/components/composer/Composer.js:84]: Are we keeping logs as comments or deleting them completely if not needed anymore?

**Review by vans1646 — CHANGES_REQUESTED** [review]
Looks already really good, but I had some questions.
Please consider them but should not be any big changes

File: app/backend/recipes/src/main/java/com/juotava/recipes/controller/RecipesController.java
Comment (MyMrMush) [reply app/backend/recipes/src/main/java/com/juotava/recipes/controller/RecipesController.java:1]: From the backend side currently all requests need authentication.
But even if we would change that, a draft should not be public so if the request was unauthorized a situation like
undefined != auth0id && draft would happen, which would then be denied as intended

File: app/frontend/src/components/composer/Composer.js (hunk 1 above)
Comment (MyMrMush) [reply app/frontend/src/components/composer/Composer.js:84]: It does not really make a difference, I will delete this one

**Pull request merged**